frontend/src/scenes/surveys/surveyLogic.test.ts

```diff
@@ -398,3 +398,406 @@ describe('single choice survey with open choice logic', () => {
+            displayThankYouMessage: true,
+            ratingButtonActiveColor: 'black',
+        },
+        created_at: '2023-10-12T06:46:32.113745Z',
+        created_by: {
+            id: 1,
+            uuid: '018aa8a6-10e8-0000-dba2-0e956f7bae38',
+            distinct_id: 'TGqg9Cn4jLkj9X87oXni9ZPBD6VbOxMtGV1GfJeB5LO',
+            first_name: 'test',
+            email: 'test@posthog.com',
+            is_email_verified: false,
+        },
+        start_date: '2023-10-12T06:46:34.482000Z',
+        end_date: null,
+        archived: false,
+        targeting_flag_filters: undefined,
+        responses_limit: null,
+    }
+
+    describe('main', () => {
+        // Single-choice question
+        it('set response-based branching for a single-choice question', async () => {
+            SURVEY.questions = [
+                {
+                    type: SurveyQuestionType.SingleChoice,
+                    choices: ['Yes', 'No'],
+                    question: 'Are you happy with our service?',
+                    description: '',
+                },
+                {
+                    type: SurveyQuestionType.Open,
+                    question: 'Glad to hear that. Tell us more!',
+                    description: '',
+                },
+                {
+                    type: SurveyQuestionType.Open,
+                    question: 'Sorry to hear that. Tell us more!',
+                    description: '',
+                },
+            ]
+
+            await expectLogic(logic, () => {
+                logic.actions.loadSurveySuccess(SURVEY)
+            }).toDispatchActions(['loadSurveySuccess'])
+
+            const questionIndex = 0
+
+            await expectLogic(logic, () => {
+                logic.actions.setQuestionBranchingType(questionIndex, SurveyQuestionBranchingType.ResponseBased)
+                logic.actions.setResponseBasedBranchingForQuestion(
+                    questionIndex,
+                    'Yes',
+                    `${SurveyQuestionBranchingType.SpecificQuestion}:1`
+                )
+                logic.actions.setResponseBasedBranchingForQuestion(
+                    questionIndex,
+                    'No',
+                    `${SurveyQuestionBranchingType.SpecificQuestion}:2`
+                )
+            })
+                .toDispatchActions([
+                    'setQuestionBranchingType',
+                    'setResponseBasedBranchingForQuestion',
+                    'setResponseBasedBranchingForQuestion',
+                ])
+                .toMatchValues({
+                    survey: partial({
+                        questions: [
+                            {
+                                ...SURVEY.questions[0],
+                                branching: {
+                                    type: SurveyQuestionBranchingType.ResponseBased,
+                                    responseValues: { Yes: 1, No: 2 },
```

Comment:
For single choice questions, I think this can be a bit problematic, given you can have multiple options with the same string, and what happens when you say, edit an option text. This probably works ok with shuffling though.

Not sure yet if we should use choiceIndex here instead, or if it's not worth the hassle at all. Can go with this for now since its flagged, and revisit once we have posthog-js running as well and see trade-offs.

Reply:
Great point, I think having multiple options with the same string is unlikely (and we can disallow that), but editing an option will indeed break this.